### PR TITLE
✨Dynamic backend: Allow the use of EC2 instance types and properly fuse their resources

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
This PR ensures that service resources in _projects_nodes_ are properly setup in case of a product where EC2 instances may be selected.

Also fixes an issue in director-v2 that was assigning the EC2 type to the label instead of the placement constraints

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py [hunk 1]
```diff
@@ -253,6 +271,86 @@ async def _get_default_pricing_and_hardware_info(
     )
 
 
+_RAM_SAFE_MARGIN_RATIO: Final[
+    float
+] = 0.1  # NOTE: machines always have less available RAM than advertised
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:276]: Is this a percent? If yes maybe reflect that in the name. If a ration, can we have a hint about the ration between what?

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py [hunk 2]
```diff
@@ -253,6 +271,86 @@ async def _get_default_pricing_and_hardware_info(
...
+            image_resources: ImageResources = node_resources[
+                DEFAULT_SINGLE_SERVICE_NAME
+            ]
+            image_resources.resources["CPU"].set_value(
+                float(selected_ec2_instance_type.cpus) - _CPUS_SAFE_MARGIN
+            )
+            image_resources.resources["RAM"].set_value(
+                int(
+                    selected_ec2_instance_type.ram
+                    - _RAM_SAFE_MARGIN_RATIO * selected_ec2_instance_type.ram
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:326]: you could also express it as `selected_ec2_instance_type.ram * 0.9` where 0.9 can be called `_SAFE_TOTAL_RAM_PERCENT`?

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py (hunk 2 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/projects/projects_api.py:326]: a percent goes from 1 - 100
a ratio from 0 - 1

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/projects/projects_api.py:276]: it is not a percent, it is a ratio

**Pull request merged**